src/modules/0.0.22/aws_s3/rpcs/s3_upload_object.ts

```diff
@@ -0,0 +1,148 @@
+import { S3, waitUntilObjectExists, _Object } from '@aws-sdk/client-s3';
+import { WaiterOptions } from '@aws-sdk/util-waiter';
+
+import { AwsS3Module } from '..';
+import { AWS, crudBuilder2, crudBuilderFormat } from '../../../../services/aws_macros';
+import { Context, RpcBase, RpcResponseObject } from '../../../interfaces';
+import { BucketObject } from '../entity';
+
+function isValidHttpUrl(path: string) {
+  let url;
+
+  try {
+    url = new URL(path);
+  } catch (_) {
+    return false;
+  }
+  return url.protocol === 'http:' || url.protocol === 'https:';
+}
+
+export class S3UploadObjectRpc extends RpcBase {
+  module: AwsS3Module;
+  outputTable = {
+    bucket: 'varchar',
+    key: 'varchar',
+    status: 'varchar',
+    response_message: 'varchar',
+  } as const;
+
+  getBucketLocation = crudBuilderFormat<S3, 'getBucketLocation', string | undefined>(
+    'getBucketLocation',
+    name => ({
+      Bucket: name,
+    }),
+    res => res?.LocationConstraint,
+  );
+
+  call = async (
+    _dbId: string,
+    _dbUser: string,
+    ctx: Context,
+    bucketName: string,
+    bucketKey: string,
+    content: string,
+    contentType: string,
+  ): Promise<RpcResponseObject<typeof this.outputTable>[]> => {
+    // we need to have bucket name, key and source
+    if (!bucketName || !bucketKey || !content || !contentType) {
+      return [
+        {
+          bucket: 'none',
+          key: 'none',
+          status: 'KO',
+          response_message:
+            'Please specify bucket name, key, content and content type of the object to upload',
+        },
+      ];
```

Comment:
Not sure if it makes sense to throw in these cases

Reply:
Yeah, I think that might be better, too.